packages/polaris-viz/src/hooks/ColorVisionA11y/useColorVisionEvents.ts

```diff
@@ -70,5 +78,5 @@ export function useColorVisionEvents(enabled = true) {
         item.removeEventListener('blur', onMouseLeave);
       });
     };
-  }, [id, enabled, hiddenIndexes]);
+  }, [id, enabled, hiddenIndexes, dimensions]);
```

Comment:
We need the event listeners to be re-added when the dimensions change so they can be applied to the updated displayed legend items.